p24/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+mod bad_lifetimes;
+fn f1(a: &mut (u32, u32), b: bool) -> &mut u32 {
+    if !b { &mut a.0 } else { &mut a.1 }
+}
+
+fn f2(a: &mut [u32], n: usize) -> &mut u32 {
+    &mut a[n]
+}
+
+fn f3(a: &mut [u32], n: usize) -> &mut u32 {
+    &mut a[a.len() - 1 - n]
```

Comment:
Note the possibility of underflow. When compiled in release mode this may cause bugs for very big `n`s close to `usize::MAX` since this computation can result in a valid index. So it's worth to check that `n` is smaller than `a.len()`.

Reply:
applied your feedback, thanks